examples/syscall-ftrace.c

```diff
@@ -0,0 +1,175 @@
+/**
+ * syscall-ftrace.c
+ * 
+ * System call "stealing" with Ftrace
+ */
+
+#include <linux/kernel.h>
+#include <linux/init.h>
+#include <linux/module.h>
+#include <linux/moduleparam.h>
+#include <linux/version.h>
+#include <linux/unistd.h>
+#include <linux/kprobes.h>
+#include <linux/sched.h>
+#include <linux/uaccess.h>
+#include <linux/slab.h>
+/** This is what we're using here. */
+#include <linux/ftrace.h>
+
+MODULE_LICENSE("GPL");
+
+#define MAX_FILENAME_SIZE 200
+
+/* UID we want to spy on - will be filled from the command line. */
+static int uid;
```

Comment:
So, do we have to give the input value every time when running this example?
If so, we should not execute this example during the checking. Then, it will be added to the non-working list (located at  `/lkmpg/.ci/non-working`).

Also, `uid` should be initialized.

Thanks.

Reply:
I followed the logic from [`syscall.c`](https://github.com/sysprog21/lkmpg/blob/9c25e59b439fcdd5fb473c9635716a1de03853b6/examples/syscall.c#L65-L66) so yes `uid` is given at `insmod` as an parameter.

Reply:
Hmm, I looked at the syscall.c again. In the syscall.c, the uid is unused variable since we removed the if-statement (check the current->uid is same as the uid) in our_sys_open(). (Original version: https://tldp.org/LDP/lkmpg/2.6/html/x978.html)

It's weird, I will fix it.

However, the uid is only used at the self-defined syscall (the use case of printk() is unimportant), which means we don't care about it when doing the checking (insert and remove the module several times, so it will test the registration only).
So, we don't have to add the syscall.c to the non-working list.
And, I strongly believe the  complicated registration of syscall.c should be tested in the checking.

Back to the PR, I roughly look at the code. Seems like it is similar to the syscall.c. So, I suggest the code here should also add the if-statement in our_sys_openat(), just like the original our_sys_open().
And, we should test the registration too.

BTW, for the initialization of uid, I just forgot the static variable will be initialized by zero. So, it's fine.

Reply:
Yeah I added the missing comparison [here](https://github.com/sysprog21/lkmpg/pull/180/commits/9b7be185b1c9dab0654bc9932bcb8e2ba871b28a).

And for `uid` initialization, I explicitly set it to zero but it should work anyways.

Reply:
Also `syscall.c` doesn't come with a user-space program in `example/other`, so I assume I don't need one here?

Reply:
Yeah, we can leave it right now.
I will consider again do we need the user programs or not.

Thanks.